src/api/app/assets/javascripts/webui2/request.js

```diff
@@ -0,0 +1,211 @@
+function updateSupersedeAndDevelPackageDisplay() {
+  if ($('#targetproject').length > 0 && $('#targetproject')[0].value.length > 2) {
+    if ($('#targetproject')[0].value === $('#sourceproject')[0].value) {
+      $('#sourceupdate-display').hide();
+      $('#sourceupdate').prop('disabled', true); // hide 'sourceupdate' from Ruby
+    } else {
+      $('#sourceupdate-display').show();
+      $('#sourceupdate').prop('disabled', false);
+    }
+    $.ajax({
+      url: $('#targetproject').data('requests-url'),
+      data: {
+        project: $('#targetproject')[0].value,
+        source_project: $('#project')[0].value, // jshint ignore:line
+        package: $('#package')[0].value,
+        types: 'submit',
+        states: ['new', 'review', 'declined']
+      },
+      success: function (data) {
+        if (data.indexOf('No requests') === -1) {
+          $('#supersede-display').show();
+          $('#supersede-requests').html(data);
+        } else {
+          $('#supersede-display').hide();
+          $('#supersede-requests').html('');
+        }
+      }
+    });
+    $.ajax({
+      url: $('#targetproject').data('develpackage-url'),
+      data: {
+        project: $('#targetproject')[0].value,
+        package: $('#package')[0].value
+      },
+      success: function (data) {
+        if (data.length > 0) {
+          $('#devel-project-warning').show();
+          $('#devel-project-name').html(data);
+        } else {
+          $('#devel-project-warning').hide();
+        }
+      }
+    });
+  }
+}
+
+function setup_request_dialog() { // jshint ignore:line
```

Comment:
now that things are copied: please use the same function naming convention in all of javascript. I know, I'm to blame for most of the snake_case in javascript - but that was before @ancorgs showed me the light :)